### PR TITLE
Http streaming

### PR DESCRIPTION
**Pull request by nsauro**
preview of http streaming..rebasing right now.

### CODE REVIEW

**Comment by nsauro** [comment]
This version is ready for review

File: colossus-tests/src/test/scala/colossus/controller/PipeSpec.scala [hunk 1]
```diff
@@ -84,7 +84,7 @@ class PipeSpec extends WordSpec with MustMatchers {
       pipe.push(1) must equal(Success(PushResult.Ok))
       val r = pipe.push(1) 
       var triggered = false
-      r mustBe an[Success[PushResult.Full]]
+      r mustBe an[Success[_]]
```
Comment (DanSimon) [inline colossus-tests/src/test/scala/colossus/controller/PipeSpec.scala]: These tests needs to be further fixed, simply checking if its a Success is not enough

File: colossus-tests/src/test/scala/colossus/controller/InputControllerSpec.scala [hunk 2]
```diff
@@ -17,7 +14,9 @@ class InputControllerSpec extends ColossusSpec {
   "Input Controller" must {
     "decode a stream message" in {
       val expected = ByteString("Hello world!")
+      println(s"size: ${expected.size.toString}")
```
Comment (DanSimon) [inline colossus-tests/src/test/scala/colossus/controller/InputControllerSpec.scala]: printlns

**Comment by DanSimon** [comment]
Looks good for now, lot of open questions though.  It would be awesome if we could figure out a way not to have two different codecs, but at least right now I don't see it.

**Pull request merged**